R/module_nested_tabs.R

```diff
@@ -48,43 +48,198 @@ ui_teal_module.default <- function(id, modules, depth = 0L) {
   stop("Modules class not supported: ", paste(class(modules), collapse = " "))
 }
 
+flatten_modules_for_dropdown <- function(modules, group_name = "") {
+  all_modules <- list()
+
+  for (i in seq_along(modules$children)) {
+    child_name <- names(modules$children)[i]
+    child <- modules$children[[i]]
+
+    if (inherits(child, "teal_modules")) {
+      nested_group_name <- if (is.null(child$label)) child_name else child$label
+      nested_modules <- flatten_modules_for_dropdown(child, nested_group_name)
+      all_modules <- c(all_modules, nested_modules)
+    } else {
+      current_group <- if (group_name != "") group_name else "Main"
+      module_info <- list(
+        id = child_name,
+        label = if (is.null(child$label)) child_name else child$label,
+        icon = icon("fas fa-chart-bar"),
+        group = current_group,
+        module_obj = child
+      )
+      all_modules[[child_name]] <- module_info
+    }
+  }
+
+  return(all_modules)
+}
+
+create_modules_dropdown <- function(ns, modules) {
+  all_modules <- flatten_modules_for_dropdown(modules)
+  grouped_modules <- split(all_modules, sapply(all_modules, function(x) x$group))
```

Comment:
Modules are flatten just to split them again. Looks like dropdown and panels could be built recursively without prior conversion. 

------

If `modules` doesn't allow nesting, maybe we should change a definition of `modules` class? `module` function could build the structure which fits your purpose:

```yaml
modules: <list>
  [1] <teal_module>
    group: "group"
    label: "label"
    ...
  [2]
    group: "...",
    label: "..."
```

Reply:
`modules()` now returns a `teal_modules` object which is a named list of `teal_module`. The names are used as shiny input id for the module selection widget.